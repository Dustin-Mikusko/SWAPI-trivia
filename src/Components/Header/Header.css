
/* * {
  border: 1px solid red;
} */
header {
  padding: .1%;
  background-color: #000;

}

.header-container {
  background-color: #000;
  height: 125px;
  color: #fff;
  display: flex;
  justify-content: space-between;
  box-shadow: 0 30px 20px -20px red;

}

.header-quote {
  margin: auto 0px;
  font-size: 2.5em;
  color: #e8c539;
  font-family: arial;
  font-style: italic;
}

.logout-button {
  width: 165px;
  height: 115px;
}

h2, p {
  margin: 0;
  font-family: 'Cormorant Garamond', serif;
}

.user-badge {
  height: 100px;
  text-align: center;
  margin: .5%;
}

.baby-yoda {
  width: 100px;
  margin: 0;
}
