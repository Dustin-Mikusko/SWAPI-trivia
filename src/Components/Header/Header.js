import React from 'react';
import './Header.css';


const Header = () => {
  return (
    <header>
    Helpful header 
    </header>
  )
}
export default Header;
