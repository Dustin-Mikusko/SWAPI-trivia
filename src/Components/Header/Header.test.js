import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe.only('Header', () => {
  const user = {
    name: "Tom Brady",
    quote: "My favorite ring is the next one.",
    rank: "beginner" 
  };


  it('should render component to match snapshot with data passed correctly', () => {
    const wrapper = shallow(<Header 
      user={user}
      logOut={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call the userLogOut function when the image is clicked', () => {
    const logOutMock = jest.fn();
    const wrapper = shallow(<Header 
      user={user}
      logOut={logOutMock}
    />);
    
    wrapper.find('Link').simulate('click');

    expect(logOutMock).toHaveBeenCalled();
  })
})
