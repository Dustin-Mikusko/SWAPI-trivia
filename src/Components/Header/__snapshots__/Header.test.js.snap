// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render component to match snapshot with data passed correctly 1`] = `
<header>
  <div
    className="header-container"
  >
    <div
      className="user-badge"
    >
      <h2>
        Tom Brady
      </h2>
      <img
        className="baby-yoda"
        src="baby-yoda.jpeg"
      />
      <p>
        beginner
      </p>
    </div>
    <h2
      className="header-quote"
    >
      My favorite ring is the next one.
    </h2>
    <Link
      onClick={[MockFunction]}
      replace={true}
      to="/"
    >
      <img
        className="logout-button"
        src="light-saber-x.jpeg"
      />
    </Link>
  </div>
</header>
`;
